metadata-ingestion/src/datahub/ingestion/source/clickhouse.py

```diff
@@ -0,0 +1,301 @@
+                actor = 'urn:li:corpuser:etl'
+                sys_time = int(time.time() * 1000)
+                schema_metadata = SchemaMetadata(
+                    schemaName=dataset_name,
+                    version=0,
+                    hash='',
+                    platform=make_data_platform_urn(platform),
+                    platformSchema=OtherSchemaClass(rawSchema=table_raw[0][0]),
+                    fields=fields,
+                    created=AuditStamp(time=sys_time,
+                                       actor=actor),
+                    lastModified=AuditStamp(time=sys_time,
+                                            actor=actor)
+                )
+                dataset_snapshot.aspects.append(schema_metadata)
+
+                mce = MetadataChangeEvent(proposedSnapshot=dataset_snapshot)
+                wu = MetadataWorkUnit(id=dataset_name,
+                                      mce=mce)
+                self.report.report_workunit(wu)
+                yield wu
+
+    def get_report(self):
+        return self.report
+
+    def close(self):
+        self._client.disconnect()
+
+
+# https://clickhouse.com/docs/en/sql-reference/data-types/
+# https://github.com/mymarilyn/clickhouse-driver/tree/master/clickhouse_driver/columns
+_field_type_mapping_clickhouse = {
+    ArrayColumn.__name__: ArrayTypeClass,
+
+    DateColumn.__name__: DateTypeClass,
+    Date32Column.__name__: DateTypeClass,
+
+    DateTimeColumn.__name__: TimeTypeClass,
+    DateTime64Column.__name__: TimeTypeClass,
+
+    DecimalColumn.__name__: NumberTypeClass,
+    Decimal32Column.__name__: NumberTypeClass,
+    Decimal64Column.__name__: NumberTypeClass,
+    Decimal128Column.__name__: NumberTypeClass,
+    Decimal256Column.__name__: NumberTypeClass,
+
+    EnumColumn.__name__: EnumTypeClass,
+    Enum8Column.__name__: EnumTypeClass,
+    Enum16Column.__name__: EnumTypeClass,
+
+    FloatColumn.__name__: NumberTypeClass,
+    Float32Column.__name__: NumberTypeClass,
+    Float64Column.__name__: NumberTypeClass,
+
+    IntColumn.__name__: NumberTypeClass,
+    UIntColumn.__name__: NumberTypeClass,
+    Int8Column.__name__: NumberTypeClass,
+    Int16Column.__name__: NumberTypeClass,
+    Int32Column.__name__: NumberTypeClass,
+    Int64Column.__name__: NumberTypeClass,
+    UInt8Column.__name__: NumberTypeClass,
+    UInt16Column.__name__: NumberTypeClass,
+    UInt32Column.__name__: NumberTypeClass,
+    UInt64Column.__name__: NumberTypeClass,
+    LargeIntColumn.__name__: NumberTypeClass,
+    Int128Column.__name__: NumberTypeClass,
+    UInt128Column.__name__: NumberTypeClass,
+    Int256Column.__name__: NumberTypeClass,
+    UInt256Column.__name__: NumberTypeClass,
+
+    IPv4Column.__name__: NumberTypeClass,
+    IPv6Column.__name__: NumberTypeClass,
+
+    LowCardinalityColumn.__name__: RecordTypeClass,
+
+    MapColumn.__name__: MapTypeClass,
+
+    String.__name__: StringTypeClass,
+    ByteString.__name__: StringTypeClass,
```

Comment:
There's a `BytesTypeClass` that could fit here—though I'm not sure what a ByteString is